### ISSUE TITLE
installing local deps using local tgz or via external https

### ISSUE THREAD

**Issue by userquin** [stackblitz/webcontainer-core#1112]
### Describe the bug

Is there any chance to allow install dependencies via local tgz files using `file:` protocol or external `https` protocol? I'm using `pnpm` as package manager (not tested with other package managers).

I'm including a new feature in Vitest repo to allow run the test in the browser running Vitest server in StackBlitz; we need a new provider for StackBlitz, that's what I'm testing with local tgz files.

I've tried with [vitest-lit-browser](https://github.com/userquin/vitest-lit-browser) gh repo using also raw https link to the tgz files, but seems blocked by CORS.

### Link to the blitz that caused the error

https://stackblitz.com/github/userquin/vitest-lit-browser

### Steps to reproduce

Just open the link, StackBlitz will not install tgz dependencies.

You can replace `vitest` and `@vitest/browser` dependencies with `https://github.com/userquin/vitest-lit-browser/raw/main/vitest-0.32.2.tgz` and `https://github.com/userquin/vitest-lit-browser/raw/main/vitest-browser-0.32.2.tgz`  respectivelly and run  `pnpm install` from the terminal.

### Expected behavior

Allow install local tgz dependencies.

### Parity with Local

- [X] I have [run the project in my local machine](https://github.com/stackblitz/webcontainer-core/blob/main/repro.md#try-locally) and I could not reproduce the issue.

### Screenshots

![image](https://github.com/stackblitz/webcontainer-core/assets/6311119/cbb09e54-8ff1-46ab-941f-48644955ed30)

### Platform

- OS: Windows 10 PRO
- Browser: Chrome, Firefox]

### Additional context

To run Vitest Browser Runner in local, `pnpm install && pnpm test`.

**Comment by jrvidal** [comment]
The tgz issue is definitely a bug, we'll fix it soon.

**Comment by userquin** [comment]
@patak-dev @jrvidal here small repo for e2e, we should fix some problem with Vitest ESM Injector and maybe add some support for routing (expose a new option to create a new iframe inside the iframe test, we need to review it).

https://github.com/userquin/vitest-vue-e2e-browser

**Comment by jrvidal** [comment]
@userquin Installing local tgz files should work now. Please feel free to open a new issue if you find anything else!

**Comment by userquin** [comment]
@jrvidal thx 

![imagen](https://github.com/stackblitz/webcontainer-core/assets/6311119/a1cf0097-1888-4319-9299-e93e2c1e629b)
